vendor/bat-native-ads/src/bat/ads/internal/account/ad_rewards/ad_rewards.cc

```diff
@@ -289,6 +289,9 @@ void AdRewards::OnGetPayments(const UrlResponse& url_response) {
     return;
   }
 
+  unreconciled_estimated_pending_rewards_ = 0.0;
+  ConfirmationsState::Get()->Save();
```

Comment:
Set `unreconciled_estimated_pending_rewards_` to `0.0` if we successfully persist the payment balances. As the issue was that `unreconciled_estimated_pending_rewards_` was only reset to `0.0` if both `GetPayment` and `GetAdGrant` endpoints were successful